src/os/bluestore/NVMEDevice.cc

```diff
@@ -550,6 +550,37 @@ static void attach_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid,
   ctx->manager->register_ctrlr(ctx->trid, ctrlr, &ctx->driver);
 }
 
+static int hex2dec(unsigned char c)
+{
+  if (isdigit(c))
+    return c - '0';
+  else if (isupper(c))
+    return c - 'A' + 10;
+  else
+    return c - 'a' + 10;
+}
+
+static int find_fisrt_bitset(const char *buf)
+{
+  int pos = 0;
+
+  if (buf[0] == '0' && ((buf[1] == 'x')
+      || (buf[1] == 'X')))
+    buf += 2;
+
+  int i = strlen(buf);
```

Comment:
nit, 1) we can just pass the `string` to `find_first_bitset()`, so we don't need to call `strlen()` for figuring out its length, 2) might want to fix the typo in the function name. 3) no need to define `pos` out of the "for" loop.

so something like would suffice:

```c++
static int find_first_bitset(const string& s)
{
  auto p = rbegin(s);
  if (s.compare(0, 2, "0x") == 0 ||
      s.compare(0, 2, "0X") == 0) {
    advance(p, 2);
  }
  for (int pos = 0; p != rend(s); ++p, pos += 4) {
    if (!isxdigit(c)) {
      return -EINVAL;
    }
    if (int val = hex2dec(*p); val != 0) {
      return pos + ffs(val);
    }
  }
  return 0;
}
```